design-doc.md

```diff
@@ -0,0 +1,137 @@
+2. No new decorators or syntax introduced
+
+#### Cons
+
+1. Ugly definition
+2. We are gradually moving away from paging from `azure-core`, and unclear if there would be a counterpart in unbranded paging, so this won't be a good idea for 3p support or non-brownfield service support
+3. No 3p support
+
+#### Todos
+
+1. Fix [this](https://github.com/Azure/typespec-azure/issues/1880) github issue
+2. Ensure support for `@nextPageOperation` in all language emitters
+
+### Adding an incomplete next link scalar type
+
+```tsp
+scalar incompleteNextLink<ParametersToReinject[]> extends url;
+
+model ListCertificateOptions {
+    includePending?: string;
+}
+
+model Page {
+    @pagedItems items: Certificate[];
+    @nextLink nextLink: incompleteNextLink<[ListCertificateOptions.includePending]>;
+}
+
+op listCertificates(...ListCertificateOptions)
+```
+
+If we found more use cases, we could elevate the definition of this scalar into either the `Azure.Core` library, or even more generic.
+
+#### Pros
+
+1. Flexible level of support. We could start with the scalar just being a local definition, but if we found other services needing this, we could elevate it up further in the library-chain
+2. Kind of fits in with the paging design that is currently in tsp.
+
+#### Cons
+
+1. Slightly more involved than getting `@nextPageOperation` to work, but not by much.
+
+#### Todos
+
+1. Have emitters recognize this definition. Look at the `armResourceIdentifier` decorator implementation from the ARM library.
+2. Potentially move it into a higher-up library.
+
+### The `@reinject` Decorator
+
+```tsp
+model Page<T> {
+  @pageItems items: string[];
+  @nextLink next: string;
+}
+op listCertificates(@reinject @query includePending?: string): Page<Certificate>;
+```
+
+`Azure.Core` paging example
+
+```tsp
+list is Azure.Core.ResourceOperations<ServiceTraits>.ResourceList<
+    Certificate,
+    {
+        @reinject @query includePending?: string;
+    },
+>;
+```
+
+#### Details
+
+- [ ] Can only be applied to optional parameters
+- [ ] Generated clients. This will be consistent with current client-side handling of `api-version` reinjection into next links
+- [ ] Usage with azure core templates? Usage with paging templates?
+
+#### Pros
+
+- Simple implementation, which is good considering our need to GA, and the fact that this definitely is more of a corner case.
+
+#### Cons
+
+- Adds another decorator
+- Definition isn't the most clear
+- Do we need to add a new decorator if we can polish up `@nextPageOperation` to work and persist that decorator?
+
+#### Todos
+
+1. Figure out where to put the `@reinject` decorator
+2. Implement the `@reinject` decorator
+3. Get emitters to understand the `@reinject` decorator
+
+## Things to Decide
+
+- [ ] design to choose
+- [ ] Should this be for azure brownfield services only? Thoughts about this being 3p behavior?
```

Comment:
This shold include 3P, at least long term, so we should have a vision of how we move to 3P.

Reply:
Sounds good. The reason I have it open is after discussing with some people on the TypeSpec team, they're not convinced it should be supported for 3p, so I feel that point is still up for debate to some, so we should definitely keep talking about it